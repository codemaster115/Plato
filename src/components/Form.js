import React from 'react';
import { push } from 'connected-react-router';
import { connect } from 'react-redux';
import styled from 'styled-components';
// import PropTypes from 'prop-types';
import Divider from '@material-ui/core/Divider';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';

import LoginField from './Login/Field';
import SignupField from './Signup/Field';

import { login } from '../ducks/login';

const Form = styled.form`
    width: 100%;
    margin-top: 8px;
`;

const StyledDivider = styled(Divider)`
    width: 100%;
    margin-left: 0px !important;
`;

const InputSection = styled.div`
    background-color: rgb(246, 249, 252) !important;
    padding: 0px 24px;
`;

const LogInBottom = styled.div`
    padding: 40px 24px 0px;
`;
const StyledButton = styled(Button)`
    height: 40px;
    font-family: "Open Sans" !important;
    font-size: 16px !important;
    margin-bottom: 10px !important;
`;

const StyledLink = styled(Link)`
    position: relative;
    left: 50%;
    transform: translateX(-50%);
    margin: 5px 0px 10px !important;
`;

class FormComponent extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
    };
  }

  UNSAFE_componentWillReceiveProps({ isAuthorized }) {
    if (isAuthorized) {
      this.props.push('./');
    }
    else {
      this.props.push('./login');
    }
  }

  onLogin = () => {    
    if(this.props.path === '/login'){
       this.props.login(this.state);
    }     
  }

  OnUserNameChange = (e) => {    
    this.setState({username: e});    
  }

  OnPasswordChange = (e) => {
    this.setState({password: e});
  }
  render() {
    return (
      <Form>
      <StyledDivider variant="middle" />     
      <InputSection>
        {
          this.props.path === "/login" ? <LoginField userNameChange={this.OnUserNameChange} passwordChange={this.OnPasswordChange}/> : <SignupField />
        }
      </InputSection>
      <StyledDivider variant="middle"/>         
      <LogInBottom>
        <StyledButton
          fullWidth
          variant="contained"
          color="primary"    
          onClick={() => {this.onLogin()}}          
        >
          {
            this.props.path === "/login" ? "Log in" : "Sign up"
          }
        </StyledButton>
        <StyledLink
          component="button"
          variant="body2"
          onClick={() => {
            this.props.path === "/login" ? this.props.push('/signup') : this.props.push('/login')
          }}
        >
          {
            this.props.path === "/login" ? "Sign up here" : "Already signed up ? Log you in"
          }         
        </StyledLink>
      </LogInBottom>              
    </Form>     
    );
  }
}

const mapStateToProps = state => (
  {
    isAuthorized: state.Login.isAuthorized,
  }
);

const mapDispatchToProps = {
  login,
  push,
}

export default connect(mapStateToProps, mapDispatchToProps)(FormComponent);
