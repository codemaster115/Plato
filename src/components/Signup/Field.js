import React from 'react';
import {
  FormControl, 
  Input, 
  InputLabel
 } from '@material-ui/core';

class ButtonComponent extends React.Component {
  render() {
    return (
      <div>
         <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="name">First Name</InputLabel>
          <Input id="first_name" name="first_name" autoFocus />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="name">Last Name</InputLabel>
          <Input id="last_name" name="last_name" />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="email">Email</InputLabel>
          <Input id="email" name="email" autoComplete="email" />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="password">Password</InputLabel>
          <Input name="password" type="password" id="password" autoComplete="current-password" />
        </FormControl>
        <FormControl margin="normal" required fullWidth>
          <InputLabel htmlFor="confirm_password">Confirm your Password</InputLabel>
          <Input name="confirm_password" type="password" id="confirm_password" autoComplete="current-password" />
        </FormControl>
      </div>
    );
  }
}

export default ButtonComponent;
