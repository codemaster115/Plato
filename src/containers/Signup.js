import React from 'react';
import { connect } from 'react-redux';
import StyledTypography from '../components/Typography';
import Form from '../components/Form';

export class SignupLayoutContainer extends React.Component {
  onValChange = e => {
    this.setState({ username: e.target.value });
  };

  render() {
    return (
      <div>
        <StyledTypography typography="Sign up to Plato Back-office" />
        <Form path={this.props.pathname} />
      </div>
    );
  }
}

SignupLayoutContainer.propTypes = {};

const mapStateToProps = state => ({
  pathname: state.router.location.pathname,
});

const mapDispatchToProps = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(SignupLayoutContainer);
