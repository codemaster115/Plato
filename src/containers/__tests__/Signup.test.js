import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import Signup, { SignupLayoutContainer } from '../Signup';
import Form from '../../components/Form';

const initialState = {};
const mockStore = configureStore();
let store;

describe('SignupLayoutContainer', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <SignupLayoutContainer />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render form', () => {
    const wrapper = shallow(<SignupLayoutContainer />);
    expect(wrapper.find(Form)).toHaveLength(1);
  });
});
