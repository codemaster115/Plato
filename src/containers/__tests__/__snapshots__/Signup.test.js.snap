// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupLayoutContainer should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <SignupLayoutContainer />
</ContextProvider>
`;
